apis/cloudbuild/v1beta1/workerpool_types.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	refv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudBuildWorkerPoolSpec defines the desired state of Instance
+type CloudBuildWorkerPoolSpec struct {
+	Name        string  `json:"name,omitempty"`
+	DisplayName string  `json:"displayName,omitempty"`
+	ResourceID  *string `json:"resourceID,omitempty"`
+	// +required
+	Parent Parent `json:"parent,omitempty"`
```

Comment:
I know... I want to try this new approach since I think it better reflects the URI and is more easy to understand.
Here's the AIP doc https://google.aip.dev/133 and I see resource to use this form in the REST doc.